spec-tests/src/test/java/org/candlepin/spec/EntitlementCertificateV3SpecTest.java

```diff
@@ -0,0 +1,146 @@
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+
+package org.candlepin.spec;
+
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import org.candlepin.ApiException;
+import org.candlepin.dto.api.v1.AsyncJobStatusDTO;
+import org.candlepin.dto.api.v1.AttributeDTO;
+import org.candlepin.dto.api.v1.ConsumerDTO;
+import org.candlepin.dto.api.v1.ContentDTO;
+import org.candlepin.dto.api.v1.OwnerDTO;
+import org.candlepin.dto.api.v1.PoolDTO;
+import org.candlepin.dto.api.v1.ProductDTO;
+import org.candlepin.resource.ConsumerApi;
+import org.candlepin.resource.OwnerApi;
+import org.candlepin.resource.OwnerContentApi;
+import org.candlepin.resource.OwnerProductApi;
+import org.candlepin.spec.bootstrap.client.ApiClient;
+import org.candlepin.spec.bootstrap.client.ApiClients;
+import org.candlepin.spec.bootstrap.client.SpecTest;
+import org.candlepin.spec.bootstrap.data.builder.Consumers;
+import org.candlepin.spec.bootstrap.data.builder.Contents;
+import org.candlepin.spec.bootstrap.data.builder.Owners;
+import org.candlepin.spec.bootstrap.data.util.CertificateUnpackUtil;
+import org.candlepin.spec.bootstrap.data.util.StringUtil;
+import org.candlepin.spec.bootstrap.data.util.X509HuffmanDecodeUtil;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import java.time.Instant;
+import java.time.ZoneOffset;
+import java.time.temporal.ChronoUnit;
+import java.util.List;
+import java.util.Random;
+
+
+@SpecTest
+public class EntitlementCertificateV3SpecTest {
+
+    private static ApiClient client;
+    private static OwnerDTO owner;
+    private OwnerApi ownerApi;
+    private OwnerContentApi ownerContentApi;
+    private OwnerProductApi ownerProductApi;
+    private ConsumerApi consumerApi;
+
+    @BeforeAll
+    public void beforeAll() throws ApiException {
+        client = ApiClients.admin();
+        ownerApi = client.owners();
+        owner = ownerApi.createOwner(Owners.random());
+        ownerContentApi = client.ownerContent();
+        ownerProductApi = client.ownerProducts();
+        consumerApi = client.consumers();
+    }
+
+    @Test
+    @DisplayName("should encode many content urls")
+    public void shouldEncodeManyContentUrls() throws Exception {
+        ProductDTO product = new ProductDTO();
+        // needs a numeric id so it will appear in the cert
+        Random rand = new Random();
+        product.setId("" + Math.abs(rand.nextInt()));
```

Comment:
I think we have utilities already for this you an use (Products.randomEng() and Products.randomSKU()). Although looking through those we probably need to switch their names (SKU products don't need numeric ids, while eng products do)